nixos/modules/services/desktops/geoclue2.nix

```diff
@@ -61,6 +186,104 @@ in
         wantedBy = [ "default.target" ];
       };
     };
+
+    environment.etc."geoclue/geoclue.conf".text = ''
+    # Configuration file for Geoclue
+
+    # Agent configuration options
+    [agent]
+
+    # Whitelist of desktop IDs (without .desktop part) of all agents we recognise,
+    # separated by a ';'.
+    whitelist=${optionalString cfg.enableDemoAgent semiColon "geoclue-demo-agent"}${semiColon "gnome-shell"}${semiColon cfg.extraWhitelistedAgents}
```

Comment:
This should not work for many reasons. I would try something like `whitelist=${let whitelist = optional cfg.enableDemoAgent "geoclue-demo-agent" ++ singleton "gnome-shell" ++ cfg.extraWhitelistedAgents; in semiColon whitelist}`